public/js/home.js

```diff
@@ -578,7 +578,19 @@ let assignmentsTable = new Tabulator("#assignmentsTable", {
             width: 40,
             align: "center",
             cellClick: function(e, cell) {
-                cell.getRow().delete();
+
+                const data = cell.getRow().getData()
+                replaceAssignmentFromID(data, {assignment_id: data["assignment_id"], placeholder: true}, selected_class_i);
+
+                new Snackbar(`You deleted ${data["name"]}`, {
+                    color: "var(--red1)",
+                    textColor: "var(--white)",
+                    buttonText: "Undo?", 
+                    buttonClick: () => {replaceAssignmentFromID({assignment_id: data["assignment_id"], placeholder: true}, data, selected_class_i);},
+                    timeout: 7500,
+                    timeoutFunction: () => {removeAssignmentFromID(data["assignment_id"], selected_class_i);},
+                    bodyClick: () => {removeAssignmentFromID(data["assignment_id"], selected_class_i);}
```

Comment:
These arrow functions can be written more simply:
```suggestion
                    buttonClick: () => replaceAssignmentFromID({ assignment_id: data["assignment_id"], placeholder: true }, data, selected_class_i),
                    timeout: 7500,
                    timeoutFunction: () => removeAssignmentFromID(data["assignment_id"], selected_class_i),
                    bodyClick: () => removeAssignmentFromID(data["assignment_id"], selected_class_i),
```